src/sentry/static/sentry/app/stores/indicatorStore.jsx

```diff
@@ -26,11 +29,11 @@ const IndicatorStore = Reflux.createStore({
     };
 
     if (options.duration) {
-      setTimeout(() => {
+      options.clearId = setTimeout(() => {
         this.remove(indicator);
       }, options.duration);
     }
-    this.items = [indicator]; // replace
+    this.items = [...this.items, indicator]; // replace
```

Comment:
Comment not valid anymore, since it's just appending. Also, I think there's is maybe a couple places in the app where we don't explicitly clear the indicator since we just rely on e.g. the loading indicator being replaced by the success/error one, so we might need to update that there. Plugins list comes to mind as an example of this.

Reply:
Yeah this is a good catch, I'll have to do a bit of refactoring to make sure this behavior is compatible because I think that pattern is used in much of the app